zeroeq/server.cpp

```diff
@@ -77,22 +77,27 @@ class Server::Impl : public detail::Sender
         }
         else
         {
-            auto reply = payload
-                             ? i->second(zmq_msg_data(&msg), zmq_msg_size(&msg))
-                             : i->second(nullptr, 0);
-            const bool hasReplyData = reply.second.ptr && reply.second.size;
+            try
+            {
+                auto reply =
+                    payload ? i->second(zmq_msg_data(&msg), zmq_msg_size(&msg))
+                            : i->second(nullptr, 0);
+                const bool hasReplyData = reply.second.ptr && reply.second.size;
 #ifdef ZEROEQ_BIGENDIAN
-            detail::byteswap(reply.first); // convert to little endian
+                detail::byteswap(reply.first); // convert to little endian
 #endif
-            if (!_send(&reply.first, sizeof(reply.first),
-                       hasReplyData ? ZMQ_SNDMORE : 0))
+                if (_send(&reply.first, sizeof(reply.first),
+                          hasReplyData ? ZMQ_SNDMORE : 0) &&
+                    hasReplyData)
+                {
+                    _send(reply.second.ptr.get(), reply.second.size, 0);
+                }
+            }
+            catch (...) // handler had exception
```

Comment:
Then we'll need to define error codes (e.g. special reply IDs) for the Client. I would vote for doing this when we have a use case.